src/NewtonSolver.cc

```diff
@@ -146,22 +145,23 @@ int NewtonSolver<Dimension, SolverType,
 
     while (1)
     {
-#ifdef DEBUG_CONVERGENCE
-        // for ( int ii = 0; ii < Dimension ; ii++ )cout<<conc[ii]<<endl;
-        // std::cout<<endl;
-
+#ifdef HAVE_OPENMP_OFFLOAD
         for (int ii = 0; ii < Dimension; ii++)
             assert(conc[ii] == conc[ii]);
```

Comment:
It is a way to check for NaN. If conc is NaN, this assert will fail.

Reply:
I never knew that, good to know!

Two questions:
1) Is there a reason this is better than using `std::isnan()`?
2) If I'm reading this correctly, it looks like the `assert` only covers the OpenMP offload case. Is there a reason it isn't needed otherwise?

Reply:
1. I guess it would be the same
2. My mistake: it is the other way around. assert() is a valid function only on CPU. Let me fix that